logs-shipper/src/main/java/com/Resource.java

```diff
@@ -43,4 +64,81 @@ public String sendLog() {
         return response;
     }
 
+    @POST
+    @Path("index")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response createTrackInJSON(RequestIndexMessage requestIndexMessage, @HeaderParam("user-agent") String userAgent) {
+        IndexRequest request = new IndexRequest("posts", "_doc");
+        request.id("" + ID++);
+
+        String msg = requestIndexMessage.getMessage();
+        if (msg == null) {
+            return Response.status(HttpURLConnection.HTTP_BAD_REQUEST).entity("The given object is null").build();
+        }
+
+        Map<String, String> jsonAsMap = new HashMap<>();
+        jsonAsMap.put("message", msg);
+        jsonAsMap.put("User-Agent", userAgent);
+
+        request.source(jsonAsMap);
+
+        client.indexAsync(request, RequestOptions.DEFAULT, new ActionListener<IndexResponse>() {
+            @Override
+            public void onResponse(IndexResponse indexResponse) {
+                System.out.println("Success Index");
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                System.out.println("Failure Index");
+                e.printStackTrace();
+            }
+        });
+
+        String result = "Success";
+        return Response.status(HttpURLConnection.HTTP_OK).entity(result).build();
+    }
+
+    @GET
+    @Path("search")
+    @Produces(MediaType.TEXT_PLAIN)
+    public String search(@QueryParam("message") String message, @QueryParam("header") String header) {
+
+        SearchRequest searchRequest = new SearchRequest("posts");
+
+        BoolQueryBuilder boolQueryBuilder = QueryBuilders.boolQuery()
+                .must(QueryBuilders.matchQuery("message", message))
+                .must(QueryBuilders.matchQuery("User-Agent", header));
+
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+        searchSourceBuilder.query(boolQueryBuilder);
+        searchRequest.source(searchSourceBuilder);
+
+        try {
+            SearchResponse searchResponse = client.search(searchRequest, RequestOptions.DEFAULT);
+
+            SearchHits hits = searchResponse.getHits();
+
+            StringBuilder builder = new StringBuilder();
+            for (SearchHit hit : hits) {
+                builder.append(hit.getSourceAsString() + "\n");
+            }
+            return builder.toString();
+
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return null;
```

Comment:
`null` means it will return 200 OK with empty body
I guess we want it respond with some error